Tests/MapboxMapsTests/Helpers/Assertions.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import XCTest
+
+func assertFirstCallParameterEqual<P, R>(
+    _ methodStub: Stub<P, R>,
+    _ firstParameter: P,
+    _ message: String = "",
+    file: StaticString = #file,
+    line: UInt = #line) where P: Equatable {
+        XCTAssertFalse(methodStub.parameters.isEmpty, message, file: file, line: line)
+        XCTAssertEqual(methodStub.parameters.first, firstParameter, message, file: file, line: line)
+    }
```

Comment:
I'm thinking about whether it might be nice to convert these to instance methods on Stub:

```
Stub.assertCall(_ index: Int, hasParameters: P, message: String = "")
```

giving usage like

```
stub.assertCall(0, hasParameters: …, message: "text here")
```

Thoughts on this direction?

Reply:
Nice idea! It looks more neat this way. But when trying it out in our tests I have some doubts:

```swift
locationChangeHandler.assertMethodCall()
```

looks more like a regular method call than anything

```swift 
assertMethodCall(locationChangeHandler)
```

While this follows the assertion pattern we use throughout the tests: `(XCT)assertX(...)`.

From this point of view I'd go with global functions, then they will be aligned with the rest of assertions and ultimately will make them easier to be parsed visually.

Reply:
```swift
        locationChangeHandler.assertMethodCall()
        XCTAssertEqual(locationChangeHandler.invocations[0].parameters.fromLocation, location)
        XCTAssertEqual(locationChangeHandler.invocations[0].parameters.toLocation.x,
                       (velocity.x * expectedVelocityAdjustmentFactor) * 0.02,
                       accuracy: 0.0000000001)
        XCTAssertEqual(locationChangeHandler.invocations[0].parameters.toLocation.y,
                       (velocity.y * expectedVelocityAdjustmentFactor) * 0.02,
                       accuracy: 0.0000000001)
```

This is the example where it breaks the flow of assertions